src/vs/platform/webContentExtractor/node/sharedWebContentExtractorService.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ISharedWebContentExtractorService } from '../common/webContentExtractor.js';
+import { URI } from '../../../base/common/uri.js';
+import { ResourceMap } from '../../../base/common/map.js';
+import { VSBuffer } from '../../../base/common/buffer.js';
+
+interface CacheEntry {
+	content: Uint8Array;
+	timestamp: number;
+}
+
+export class SharedWebContentExtractorService implements ISharedWebContentExtractorService {
+	_serviceBrand: undefined;
+	private _webContentsCache = new ResourceMap<CacheEntry>();
+	private readonly _cacheDuration = 24 * 60 * 60 * 1000; // 1 day in milliseconds
+
+	private isExpired(entry: CacheEntry): boolean {
+		return Date.now() - entry.timestamp > this._cacheDuration;
+	}
+
+	extractUrls(uris: URI): Promise<Uint8Array> {
+		return this.doExtractImage(uris);
+	}
+
+	private async doExtractImage(uri: URI): Promise<Uint8Array> {
+		const cached = this._webContentsCache.get(uri);
```

Comment:
Is the caching worth it? I'd expect that outside of testing, you really won't have hits that often. There's no upper limit on the cache so I'm concerned about it growing too big if a user is downloading a lot of images

Maybe you can cache on the caller side just to make sure it doesn't get re-downloaded if you paste, unto, then paste again? That should only require keeping the most recent image cached

Reply:
was caching since Tyler was doing something similar, but tbh it's probably not needed yeah